### DISCUSSION TITLE
How do I write a swagger endpoint where I can delay the consumption of the body in its parameters?

### DISCUSSION THREAD

**Discussion by seang-godaddy** [go-swagger/go-swagger#2709] [Q&A] [answered]
I need to define a swagger endpoint that needs to take in media types text and zip. Instead of having the generated package consume it and reassign it into separate types such as string, can I get the io.Readcloser passed through directly?

**Comment by casualjim** [comment, marked as answer]
You will get a stream, when you define the parameter as 

```yaml
type: string
format: binary
```
